### ISSUE TITLE
[Feature request] redirect_uri pattern

### ISSUE THREAD

**Issue by akvadrako** [babelouest/glewlwyd#177]
**Is your feature request related to a problem? Please describe.**

When I go to `myapp/path/abc` and I need login, I need to supply a dynamic `redirect_uri`. However every exact Redirect URI need to be supplied to glewlwyd statically.

**Describe the solution you'd like**
I would like the redirect uri to be a pattern, for example a regex.

**Additional context**
I am using glewlwyd with `caddy-auth-jwt`, which only supports `redirect_url` (*L* instead of *I*), which is also an issue.

**Comment by babelouest** [comment]
Hello @akvadrako , thanks for the feature requests.

TL;DR: I won't be able to add those features in Glewlwyd, too much bugs and security issues can raise from those.

### Redirect uri to be a pattern

It's a bad idea to allow patterns in redirect uris, it can easily lead to client impersonations.

Also, in the recent [OAuth2.1 specification](https://oauth.net/2.1/), this has been explicitly forbidden: `Redirect URIs must be compared using exact string matching`

### Supports redirect_url property as redirect_uri

That's a bad idea as well, the standard specifies `redirect_uri`. Adding more property keys for the same purpose could lead to a whole new world of bugs and security issues:
- I'd have to rewrite all the tests to handle `redirect_url` as well
- What if a client uses `redirect_uri` and `redirect_url` at the same time?
- What if a client already uses `redirect_url` for another purpose?
- etc.

### My advice (if you allow me)

Contact the maintainer of the plugin to fix it.

- The `state` parameter exists for the client to link oauth2 requests and responses. Instead of having a dynamic `redirect_uri`, i.e. with a parameter in it, like `https://client.example.com/token=xxx`, it can use the state to pass this parameter, like `https://client.example.com/?state=xxx'. This is allowed and highly recommended by the standard.

- The `redirect_uri` parameter should be used by the client instead of `redirect_url`. A quick look at the source code of your plugins shows only [one place](https://github.com/greenpau/caddy-auth-jwt/blob/main/pkg/auth/manager.go#L115) where this parameter is named

**Comment by akvadrako** [comment]
Okay, thanks for the detailed reply.

**Comment by babelouest** [comment]
No problem, I'm closing this issue then.
